contrib/node/src/python/pants/contrib/node/subsystems/resolvers/npm_resolver.py

```diff
@@ -137,9 +141,12 @@ def _link_source_dependencies(self, node_task, target, results_dir, node_paths,
 
       # Apply node-scoping rules if applicable
       node_scope = dep.payload.node_scope or node_task.node_distribution.node_scope
-      dep_package_name = self._scoped_package_name(node_task, dep.package_name, node_scope)
+      dep_package_name = self._scoped_package_name(node_task, package_name, node_scope)
       # Symlink each target
-      dep_path = node_paths.node_path(dep)
+      if isinstance(node_paths, NodePathsLocal):
```

Comment:
It's not clear why `NodePathsLocal` is relevant to whether we use `concrete_derived_from` here... a comment would be good.

Reply:
This check may no longer be necessary if we store both the `node_thrift_library` and code-gen `node_module` targets in the node_paths now.